src/main/java/com/blackducksoftware/integration/hub/sonar/component/HubVulnerableComponentGatherer.java

```diff
@@ -100,14 +101,12 @@ private void mapMatchedFilesToComponents(Map<String, Set<ProjectVersionComponent
         if (components != null && !components.isEmpty()) {
             String prevName = "";
             for (ProjectVersionComponentView component : components) {
-                RiskProfileView securityRiskProfile = component.getSecurityRiskProfile();
-                if (null != securityRiskProfile && null != securityRiskProfile.getCounts() && !securityRiskProfile.getCounts().isEmpty()) {
+                if (hasSecurityRisk(component)) {
```

Comment:
Rather than checking all of the components in the bom, why not get the vulnerable-components from the project-version initially?